### PR TITLE
wallet-rpc: add on_restore_deterministic, auto_refresh, generate_from_keys, et_log_level/set_log_categories.

### PR DESCRIPTION
**Pull request by SmajeNz0**

### CODE REVIEW

File: src/wallet/wallet_rpc_server_commands_defs.h [hunk 1]
```diff
@@ -48,7 +48,7 @@
 // advance which version they will stop working with
 // Don't go over 32767 for any of these
 #define WALLET_RPC_VERSION_MAJOR 1
-#define WALLET_RPC_VERSION_MINOR 7
+#define WALLET_RPC_VERSION_MINOR 11
```
Comment (malbit) [inline src/wallet/wallet_rpc_server_commands_defs.h:51]: Why did you bump it so high? 
That state only rpc major.minor version and do not take Hard-Fork version at all

File: src/wallet/wallet_rpc_server.cpp [hunk 2]
```diff
@@ -57,6 +57,8 @@ using namespace epee;
 #undef ARQMA_DEFAULT_LOG_CATEGORY
 #define ARQMA_DEFAULT_LOG_CATEGORY "wallet.rpc"
```
Comment (malbit) [inline src/wallet/wallet_rpc_server.cpp:59]: I would increase refresh period up to 60 seconds. 20 seconds might be too often and generates high net usage.

File: src/wallet/wallet_rpc_server_commands_defs.h (hunk 1 above)
Comment (SmajeNz0) [reply src/wallet/wallet_rpc_server_commands_defs.h:51]: // When making *any* change here, bump minor
// If the change is incompatible, then bump major and set minor to 0
// This ensures WALLET_RPC_VERSION always increases, that every change
// has its own version, and that clients can just test major to see
// whether they can talk to a given wallet without having to know in
// advance which version they will stop working with
// Don't go over 32767 for any of these

Comment (SmajeNz0) [reply src/wallet/wallet_rpc_server_commands_defs.h:51]: 4 commits -> 4 bumps.

Comment (malbit) [reply src/wallet/wallet_rpc_server_commands_defs.h:51]: Good enough for me then :)

**Review by malbit — APPROVED** [review]
approved

File: src/wallet/wallet_rpc_server.cpp (hunk 2 above)
Comment (SmajeNz0) [reply src/wallet/wallet_rpc_server.cpp:59]: It  could be changed with a RPC request.

**Review by ArqTras — COMMENTED** [review]
approved

**Review by ArqTras — APPROVED** [review]

**Pull request merged**